### ISSUE TITLE
IllegalStateException: expecting index, was header in CycleNotFoundTest

### ISSUE THREAD

**Issue by alamar** [OpenHFT/Chronicle-Queue#998]
#996 has unmasked the following exception which will very rarely occur in `net.openhft.chronicle.queue.CycleNotFoundTest#tailerCycleNotFoundTest` (on TC and in IDEA)

```
 Caused by: java.lang.IllegalStateException: expecting index, was header, bytes: 
  00000000             b9 06 68 65  61 64 65 72 b6 08 53 43     ··he ader··SC
  00000010 51 53 74 6f 72 65 82 56  00 00 00 8f 8f 8f 8f 8d QStore·V ········
  00000020 02 00 00 00 00 00 00 00  02 00 00 00 00 00 00 00 ········ ········
  00000030 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00 ········ ········
  00000040 b6 0c 53 43 51 53 49 6e  64 65 78 69 6e 67 82 1d ··SCQSIn dexing··
  00000050 00 00 00 a2 00 80 04 a7  00 00 00 00 00 00 00 00 ········ ········
  00000060 8e 02 00 00 00 00 00 a7  00 00 00 00 00 00 00 00 ········ ········
  00000070 01                                               ·
    at net.openhft.chronicle.queue.impl.single.SCQIndexing.array(SCQIndexing.java:199)
    at net.openhft.chronicle.queue.impl.single.SCQIndexing.getIndex2index(SCQIndexing.java:579)
    at net.openhft.chronicle.queue.impl.single.SCQIndexing.moveToIndex0(SCQIndexing.java:278)
    at net.openhft.chronicle.queue.impl.single.SCQIndexing.moveToIndex(SCQIndexing.java:254)
    at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueStore.moveToIndexForRead(SingleChronicleQueueStore.java:228)
    at net.openhft.chronicle.queue.impl.single.StoreTailer.moveToIndexResult0(StoreTailer.java:668)
    at net.openhft.chronicle.queue.impl.single.StoreTailer.moveToIndexInternal(StoreTailer.java:739)
    at net.openhft.chronicle.queue.impl.single.StoreTailer.next0(StoreTailer.java:294)
    at net.openhft.chronicle.queue.impl.single.StoreTailer.readingDocument0(StoreTailer.java:215)
    at net.openhft.chronicle.queue.impl.single.StoreTailer.readingDocument(StoreTailer.java:190)
    at net.openhft.chronicle.queue.impl.single.StoreTailer.readingDocument(StoreTailer.java:134)
    at net.openhft.chronicle.queue.CycleNotFoundTest.lambda$tailerCycleNotFoundTest$0(CycleNotFoundTest.java:65)
```

**Comment by alamar** [comment]
https://teamcity.chronicle.software/buildConfiguration/Chronicle_BuildAll_OpenHFTJava8/647206

**Comment by alamar** [comment]
[tailerCycleNotFoundTest.gz](https://github.com/OpenHFT/Chronicle-Queue/files/7978671/tailerCycleNotFoundTest.gz)
Uploaded is the queue from this test. It is corrupted: DumpMain can read it, but scanning it backwards resulting in:
`readCheckOffset0 failed. Offset: 524668 + adding: 4 > limit: 524669 (given: false)`

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.22ea16](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.22ea16), [BOM-2.22ea106](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea106)
